Build failure with latest master

Hi,
  I'm trying to build the project on my machine but am failing at the compilation stage (log attached).
I'm trying to add this compositor to Gentoo linux and would appreciate the help, thanks!

```
>>> Configuring source in /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999 ...
meson setup --buildtype plain --libdir lib64 --localstatedir /var/lib --prefix /usr --sysconfdir /etc --wrap-mode nodownload --build.pkg-config-path /usr/share/pkgconfig --pkg-config-path /usr/share/pkgconfig --native-file /var/tmp/portage/gui-wm/labwc-9999/temp/meson.x86_64-pc-linux-gnu.amd64.ini -Dxwayland=enabled /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999 /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build
DEPRECATION: c_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: c_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: fortran_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: fortran_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objc_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objc_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objcpp_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objcpp_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
The Meson build system
Version: 0.56.2
Source dir: /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999
Build dir: /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build
Build type: native build
Project name: labwc
Project version: undefined
C compiler for the host machine: x86_64-pc-linux-gnu-gcc (gcc 10.2.0 "x86_64-pc-linux-gnu-gcc (Gentoo 10.2.0-r5 p6) 10.2.0")
C linker for the host machine: x86_64-pc-linux-gnu-gcc ld.bfd 2.35.2
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wno-unused-parameter: YES
Compiler for C supports arguments -Wundef: YES
Neither a subproject directory nor a wlroots.wrap file was found.
Subproject  wlroots is buildable: NO (disabling)
Found pkg-config: /usr/bin/x86_64-pc-linux-gnu-pkg-config (1.7.3)
Run-time dependency wlroots found: YES 0.12.0
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
Run-time dependency wayland-server found: YES 1.19.0
Run-time dependency wayland-protocols found: YES 1.20
Run-time dependency xkbcommon found: YES 1.0.3
Run-time dependency xcb found: YES 1.14
Run-time dependency libxml-2.0 found: YES 2.9.10
Run-time dependency glib-2.0 found: YES 2.66.7
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pangocairo found: YES 1.42.4
Run-time dependency libinput found: YES 1.17.0
Run-time dependency pixman-1 found: YES 0.40.0
Library m found: YES
Program wayland-scanner found: YES (/usr/bin/wayland-scanner)
Configuring config.h using configuration
Program pandoc found: NO
Build targets in project: 2

labwc undefined

  Subprojects
    wlroots: NO Neither a subproject directory nor a wlroots.wrap file was found.

Found ninja-1.10.2 at /usr/bin/ninja
>>> Source configured.
>>> Compiling source in /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999 ...
ninja -v -j8 -l0 -C /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build
ninja: Entering directory `/var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build'
[1/44] /usr/bin/wayland-scanner private-code ../labwc-9999/protocols/wlr-layer-shell-unstable-v1.xml protocols/libserver_protos.a.p/wlr-layer-shell-unstable-v1-protocol.c
[2/44] /usr/bin/wayland-scanner server-header ../labwc-9999/protocols/wlr-layer-shell-unstable-v1.xml protocols/libserver_protos.a.p/wlr-layer-shell-unstable-v1-protocol.h
[3/44] /usr/bin/wayland-scanner private-code //usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml protocols/libserver_protos.a.p/xdg-shell-protocol.c
[4/44] /usr/bin/wayland-scanner server-header ../labwc-9999/protocols/wlr-layer-shell-unstable-v1.xml labwc.p/wlr-layer-shell-unstable-v1-protocol.h
[5/44] /usr/bin/wayland-scanner server-header //usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml protocols/libserver_protos.a.p/xdg-shell-protocol.h
[6/44] /usr/bin/wayland-scanner server-header //usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml labwc.p/xdg-shell-protocol.h
[7/44] x86_64-pc-linux-gnu-gcc -Iprotocols/libserver_protos.a.p -Iprotocols -I../labwc-9999/protocols -I/usr/lib64/libffi/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -fPIC -MD -MQ protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o -MF protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o.d -o protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o -c protocols/libserver_protos.a.p/wlr-layer-shell-unstable-v1-protocol.c
[8/44] x86_64-pc-linux-gnu-gcc -Iprotocols/libserver_protos.a.p -Iprotocols -I../labwc-9999/protocols -I/usr/lib64/libffi/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -fPIC -MD -MQ protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o -MF protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o.d -o protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o -c protocols/libserver_protos.a.p/xdg-shell-protocol.c
[9/44] rm -f protocols/libserver_protos.a && x86_64-pc-linux-gnu-ar csrD protocols/libserver_protos.a protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o
[10/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_deco.c.o -MF labwc.p/src_deco.c.o.d -o labwc.p/src_deco.c.o -c ../labwc-9999/src/deco.c
FAILED: labwc.p/src_deco.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_deco.c.o -MF labwc.p/src_deco.c.o.d -o labwc.p/src_deco.c.o -c ../labwc-9999/src/deco.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/deco.c:9:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[11/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_interactive.c.o -MF labwc.p/src_interactive.c.o.d -o labwc.p/src_interactive.c.o -c ../labwc-9999/src/interactive.c
FAILED: labwc.p/src_interactive.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_interactive.c.o -MF labwc.p/src_interactive.c.o.d -o labwc.p/src_interactive.c.o -c ../labwc-9999/src/interactive.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/interactive.c:1:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[12/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_action.c.o -MF labwc.p/src_action.c.o.d -o labwc.p/src_action.c.o -c ../labwc-9999/src/action.c
FAILED: labwc.p/src_action.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_action.c.o -MF labwc.p/src_action.c.o.d -o labwc.p/src_action.c.o -c ../labwc-9999/src/action.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/action.c:4:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[13/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_damage.c.o -MF labwc.p/src_damage.c.o.d -o labwc.p/src_damage.c.o -c ../labwc-9999/src/damage.c
FAILED: labwc.p/src_damage.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_damage.c.o -MF labwc.p/src_damage.c.o.d -o labwc.p/src_damage.c.o -c ../labwc-9999/src/damage.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/damage.c:1:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[14/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_desktop.c.o -MF labwc.p/src_desktop.c.o.d -o labwc.p/src_desktop.c.o -c ../labwc-9999/src/desktop.c
FAILED: labwc.p/src_desktop.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_desktop.c.o -MF labwc.p/src_desktop.c.o.d -o labwc.p/src_desktop.c.o -c ../labwc-9999/src/desktop.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/desktop.c:3:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[15/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_cursor.c.o -MF labwc.p/src_cursor.c.o.d -o labwc.p/src_cursor.c.o -c ../labwc-9999/src/cursor.c
FAILED: labwc.p/src_cursor.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_cursor.c.o -MF labwc.p/src_cursor.c.o.d -o labwc.p/src_cursor.c.o -c ../labwc-9999/src/cursor.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/cursor.c:2:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[16/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_keyboard.c.o -MF labwc.p/src_keyboard.c.o.d -o labwc.p/src_keyboard.c.o -c ../labwc-9999/src/keyboard.c
FAILED: labwc.p/src_keyboard.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_keyboard.c.o -MF labwc.p/src_keyboard.c.o.d -o labwc.p/src_keyboard.c.o -c ../labwc-9999/src/keyboard.c
In file included from ../labwc-9999/include/labwc.h:19,
                 from ../labwc-9999/src/keyboard.c:2:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
[17/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_layers.c.o -MF labwc.p/src_layers.c.o.d -o labwc.p/src_layers.c.o -c ../labwc-9999/src/layers.c
FAILED: labwc.p/src_layers.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_layers.c.o -MF labwc.p/src_layers.c.o.d -o labwc.p/src_layers.c.o -c ../labwc-9999/src/layers.c
In file included from ../labwc-9999/src/layers.c:14:
/usr/include/wlr/types/wlr_layer_shell_v1.h:52:52: error: field â€˜keyboard_interactiveâ€™ has incomplete type
   52 |  enum zwlr_layer_surface_v1_keyboard_interactivity keyboard_interactive;
      |                                                    ^~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

It progressed further :D but still crashed D:
```
>>> Configuring source in /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999 ...
meson setup --buildtype plain --libdir lib64 --localstatedir /var/lib --prefix /usr --sysconfdir /etc --wrap-mode nodownload --build.pkg-config-path /usr/share/pkgconfig --pkg-config-path /usr/share/pkgconfig --native-file /var/tmp/portage/gui-wm/labwc-9999/temp/meson.x86_64-pc-linux-gnu.amd64.ini -Dxwayland=enabled /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999 /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build
DEPRECATION: c_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: c_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: fortran_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: fortran_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objc_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objc_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objcpp_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: objcpp_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
The Meson build system
Version: 0.56.2
Source dir: /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999
Build dir: /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build
Build type: native build
Project name: labwc
Project version: undefined
C compiler for the host machine: x86_64-pc-linux-gnu-gcc (gcc 10.2.0 "x86_64-pc-linux-gnu-gcc (Gentoo 10.2.0-r5 p6) 10.2.0")
C linker for the host machine: x86_64-pc-linux-gnu-gcc ld.bfd 2.35.2
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wno-unused-parameter: YES
Compiler for C supports arguments -Wundef: YES
Neither a subproject directory nor a wlroots.wrap file was found.
Subproject  wlroots is buildable: NO (disabling)
Found pkg-config: /usr/bin/x86_64-pc-linux-gnu-pkg-config (1.7.3)
Run-time dependency wlroots found: YES 0.12.0
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
Run-time dependency wayland-server found: YES 1.19.0
Run-time dependency wayland-protocols found: YES 1.20
Run-time dependency xkbcommon found: YES 1.0.3
Run-time dependency xcb found: YES 1.14
Run-time dependency libxml-2.0 found: YES 2.9.10
Run-time dependency glib-2.0 found: YES 2.66.7
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pangocairo found: YES 1.42.4
Run-time dependency libinput found: YES 1.17.0
Run-time dependency pixman-1 found: YES 0.40.0
Library m found: YES
Program wayland-scanner found: YES (/usr/bin/wayland-scanner)
Configuring config.h using configuration
Program pandoc found: NO
Build targets in project: 2

labwc undefined

  Subprojects
    wlroots: NO Neither a subproject directory nor a wlroots.wrap file was found.

Found ninja-1.10.2 at /usr/bin/ninja
>>> Source configured.
>>> Compiling source in /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999 ...
ninja -v -j8 -l0 -C /var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build
ninja: Entering directory `/var/tmp/portage/gui-wm/labwc-9999/work/labwc-9999-build'
[1/44] /usr/bin/wayland-scanner private-code //usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml protocols/libserver_protos.a.p/xdg-shell-protocol.c
[2/44] /usr/bin/wayland-scanner private-code ../labwc-9999/protocols/wlr-layer-shell-unstable-v1.xml protocols/libserver_protos.a.p/wlr-layer-shell-unstable-v1-protocol.c
[3/44] /usr/bin/wayland-scanner server-header //usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml protocols/libserver_protos.a.p/xdg-shell-protocol.h
[4/44] /usr/bin/wayland-scanner server-header ../labwc-9999/protocols/wlr-layer-shell-unstable-v1.xml protocols/libserver_protos.a.p/wlr-layer-shell-unstable-v1-protocol.h
[5/44] /usr/bin/wayland-scanner server-header //usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml labwc.p/xdg-shell-protocol.h
[6/44] /usr/bin/wayland-scanner server-header ../labwc-9999/protocols/wlr-layer-shell-unstable-v1.xml labwc.p/wlr-layer-shell-unstable-v1-protocol.h
[7/44] x86_64-pc-linux-gnu-gcc -Iprotocols/libserver_protos.a.p -Iprotocols -I../labwc-9999/protocols -I/usr/lib64/libffi/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -fPIC -MD -MQ protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o -MF protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o.d -o protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o -c protocols/libserver_protos.a.p/xdg-shell-protocol.c
[8/44] x86_64-pc-linux-gnu-gcc -Iprotocols/libserver_protos.a.p -Iprotocols -I../labwc-9999/protocols -I/usr/lib64/libffi/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -fPIC -MD -MQ protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o -MF protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o.d -o protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o -c protocols/libserver_protos.a.p/wlr-layer-shell-unstable-v1-protocol.c
[9/44] rm -f protocols/libserver_protos.a && x86_64-pc-linux-gnu-ar csrD protocols/libserver_protos.a protocols/libserver_protos.a.p/meson-generated_xdg-shell-protocol.c.o protocols/libserver_protos.a.p/meson-generated_wlr-layer-shell-unstable-v1-protocol.c.o
[10/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_cursor.c.o -MF labwc.p/src_cursor.c.o.d -o labwc.p/src_cursor.c.o -c ../labwc-9999/src/cursor.c
[11/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_deco.c.o -MF labwc.p/src_deco.c.o.d -o labwc.p/src_deco.c.o -c ../labwc-9999/src/deco.c
[12/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_damage.c.o -MF labwc.p/src_damage.c.o.d -o labwc.p/src_damage.c.o -c ../labwc-9999/src/damage.c
[13/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_desktop.c.o -MF labwc.p/src_desktop.c.o.d -o labwc.p/src_desktop.c.o -c ../labwc-9999/src/desktop.c
[14/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_interactive.c.o -MF labwc.p/src_interactive.c.o.d -o labwc.p/src_interactive.c.o -c ../labwc-9999/src/interactive.c
[15/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_action.c.o -MF labwc.p/src_action.c.o.d -o labwc.p/src_action.c.o -c ../labwc-9999/src/action.c
[16/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_keyboard.c.o -MF labwc.p/src_keyboard.c.o.d -o labwc.p/src_keyboard.c.o -c ../labwc-9999/src/keyboard.c
[17/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_layers.c.o -MF labwc.p/src_layers.c.o.d -o labwc.p/src_layers.c.o -c ../labwc-9999/src/layers.c
[18/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_seat.c.o -MF labwc.p/src_seat.c.o.d -o labwc.p/src_seat.c.o -c ../labwc-9999/src/seat.c
[19/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_main.c.o -MF labwc.p/src_main.c.o.d -o labwc.p/src_main.c.o -c ../labwc-9999/src/main.c
[20/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_osd.c.o -MF labwc.p/src_osd.c.o.d -o labwc.p/src_osd.c.o -c ../labwc-9999/src/osd.c
[21/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_view.c.o -MF labwc.p/src_view.c.o.d -o labwc.p/src_view.c.o -c ../labwc-9999/src/view.c
[22/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_xdg.c.o -MF labwc.p/src_xdg.c.o.d -o labwc.p/src_xdg.c.o -c ../labwc-9999/src/xdg.c
../labwc-9999/src/xdg.c: In function â€˜xdg_toplevel_view_for_each_popupâ€™:
../labwc-9999/src/xdg.c:292:2: warning: implicit declaration of function â€˜wlr_xdg_surface_for_each_popupâ€™; did you mean â€˜wlr_xdg_surface_for_each_surfaceâ€™? [-Wimplicit-function-declaration]
  292 |  wlr_xdg_surface_for_each_popup(view->xdg_surface, iterator, data);
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |  wlr_xdg_surface_for_each_surface
[23/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_server.c.o -MF labwc.p/src_server.c.o.d -o labwc.p/src_server.c.o -c ../labwc-9999/src/server.c
FAILED: labwc.p/src_server.c.o
x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_server.c.o -MF labwc.p/src_server.c.o.d -o labwc.p/src_server.c.o -c ../labwc-9999/src/server.c
../labwc-9999/src/server.c: In function â€˜server_initâ€™:
../labwc-9999/src/server.c:93:20: error: too many arguments to function â€˜wlr_backend_autocreateâ€™
   93 |  server->backend = wlr_backend_autocreate(server->wl_display, NULL);
      |                    ^~~~~~~~~~~~~~~~~~~~~~
In file included from ../labwc-9999/include/labwc.h:11,
                 from ../labwc-9999/src/server.c:12:
/usr/include/wlr/backend.h:35:21: note: declared here
   35 | struct wlr_backend *wlr_backend_autocreate(struct wl_display *display);
      |                     ^~~~~~~~~~~~~~~~~~~~~~
[24/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_output.c.o -MF labwc.p/src_output.c.o.d -o labwc.p/src_output.c.o -c ../labwc-9999/src/output.c
[25/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_common_buf.c.o -MF labwc.p/src_common_buf.c.o.d -o labwc.p/src_common_buf.c.o -c ../labwc-9999/src/common/buf.c
[26/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_theme.c.o -MF labwc.p/src_theme.c.o.d -o labwc.p/src_theme.c.o -c ../labwc-9999/src/theme.c
[27/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_xwayland.c.o -MF labwc.p/src_xwayland.c.o.d -o labwc.p/src_xwayland.c.o -c ../labwc-9999/src/xwayland.c
[28/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_xwayland-unmanaged.c.o -MF labwc.p/src_xwayland-unmanaged.c.o.d -o labwc.p/src_xwayland-unmanaged.c.o -c ../labwc-9999/src/xwayland-unmanaged.c
[29/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_common_dir.c.o -MF labwc.p/src_common_dir.c.o.d -o labwc.p/src_common_dir.c.o -c ../labwc-9999/src/common/dir.c
[30/44] x86_64-pc-linux-gnu-gcc -Ilabwc.p -I. -I../labwc-9999 -Iinclude -I../labwc-9999/include -I/usr/lib64/libffi/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -I/usr/include/uuid -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-enum-compare -Wno-unused-parameter -Wundef -march=znver1 -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=8 -fuse-linker-plugin -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -Wl,-O1 -Wl,--as-needed -pthread -MD -MQ labwc.p/src_common_font.c.o -MF labwc.p/src_common_font.c.o.d -o labwc.p/src_common_font.c.o -c ../labwc-9999/src/common/font.c
ninja: build stopped: subcommand failed.

```

--- a/include/labwc.h
+++ b/include/labwc.h
@@ -155,7 +155,7 @@ enum deco_part {
 struct view_impl {
 	void (*configure)(struct view *view, struct wlr_box geo);
 	void (*close)(struct view *view);
-	void (*for_each_popup)(struct view *view,
+	void (*for_each_popup_surface)(struct view *view,
 		wlr_surface_iterator_func_t iterator, void *data);
 	void (*for_each_surface)(struct view *view,
 		wlr_surface_iterator_func_t iterator, void *data);
@@ -296,7 +296,7 @@ void view_unminimize(struct view *view);
 void view_maximize(struct view *view, bool maximize);
 void view_for_each_surface(struct view *view,
 	wlr_surface_iterator_func_t iterator, void *user_data);
-void view_for_each_popup(struct view *view,
+void view_for_each_popup_surface(struct view *view,
 	wlr_surface_iterator_func_t iterator, void *data);
 
 void desktop_focus_view(struct seat *seat, struct view *view);

--- a/meson.build
+++ b/meson.build
@@ -41,7 +41,7 @@ if wlroots_proj.found()
   wlroots_conf = wlroots_proj.get_variable('conf_data')
   wlroots_has_xwayland = wlroots_conf.get('WLR_HAS_XWAYLAND') == 1
 else
-  wlroots       = dependency('wlroots', version: '>= 0.11.0')
+  wlroots       = dependency('wlroots', version: '>= 0.13.0')
   wlroots_has_xwayland = cc.get_define('WLR_HAS_XWAYLAND', prefix: '#include <wlr/config.h>', dependencies: wlroots) == '1'
 endif
 wayland_server  = dependency('wayland-server')

--- a/src/menu/menu.c
+++ b/src/menu/menu.c
@@ -2,6 +2,7 @@
 #include <assert.h>
 #include <cairo/cairo.h>
 #include <ctype.h>
+#include <drm_fourcc.h>
 #include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <pango/pangocairo.h>
@@ -61,7 +62,7 @@ texture_create(struct server *server, struct wlr_box *geo, const char *text,
 	cairo_surface_flush(surf);
 	unsigned char *data = cairo_image_surface_get_data(surf);
 	struct wlr_texture *texture = wlr_texture_from_pixels(server->renderer,
-		WL_SHM_FORMAT_ARGB8888, cairo_image_surface_get_stride(surf),
+		DRM_FORMAT_ARGB8888, cairo_image_surface_get_stride(surf),
 		geo->width, geo->height, data);
 
 	cairo_destroy(cairo);

--- a/src/output.c
+++ b/src/output.c
@@ -223,7 +223,7 @@ output_view_for_each_surface(struct output *output, struct view *view,
 }
 
 void
-output_view_for_each_popup(struct output *output, struct view *view,
+output_view_for_each_popup_surface(struct output *output, struct view *view,
 		surface_iterator_func_t iterator, void *user_data)
 {
 	struct surface_iterator_data data = {
@@ -236,7 +236,7 @@ output_view_for_each_popup(struct output *output, struct view *view,
 
 	wlr_output_layout_output_coords(output->server->output_layout,
 		output->wlr_output, &data.ox, &data.oy);
-	view_for_each_popup(view, output_for_each_surface_iterator, &data);
+	view_for_each_popup_surface(view, output_for_each_surface_iterator, &data);
 }
 
 /* for sending frame done */
@@ -569,25 +569,13 @@ render_view_toplevels(struct view *view, struct output *output,
 }
 
 static void
-render_popup_iterator(struct output *output, struct wlr_surface *surface,
-		struct wlr_box *box, void *data)
-{
-	/* Render this popup's surface */
-	render_surface_iterator(output, surface, box, data);
-
-	/* Render this popup's child toplevels */
-	output_surface_for_each_surface(output, surface, box->x, box->y,
-		render_surface_iterator, data);
-}
-
-static void
 render_view_popups(struct view *view, struct output *output,
 		pixman_region32_t *damage)
 {
 	struct render_data data = {
 		.damage = damage,
 	};
-	output_view_for_each_popup(output, view, render_popup_iterator, &data);
+	output_view_for_each_popup_surface(output, view, render_surface_iterator, &data);
 }
 
 void

--- a/src/server.c
+++ b/src/server.c
@@ -90,7 +90,7 @@ server_init(struct server *server)
 	 * backend based on the current environment, such as opening an x11
 	 * window if an x11 server is running.
 	 */
-	server->backend = wlr_backend_autocreate(server->wl_display, NULL);
+	server->backend = wlr_backend_autocreate(server->wl_display);
 	if (!server->backend) {
 		wlr_log(WLR_ERROR, "unable to create backend");
 		exit(EXIT_FAILURE);

--- a/src/view.c
+++ b/src/view.c
@@ -86,12 +86,12 @@ view_for_each_surface(struct view *view, wlr_surface_iterator_func_t iterator,
 }
 
 void
-view_for_each_popup(struct view *view, wlr_surface_iterator_func_t iterator,
+view_for_each_popup_surface(struct view *view, wlr_surface_iterator_func_t iterator,
 		void *data)
 {
-	if (!view->impl->for_each_popup) {
+	if (!view->impl->for_each_popup_surface) {
 		return;
 	}
-	view->impl->for_each_popup(view, iterator, data);
+	view->impl->for_each_popup_surface(view, iterator, data);
 }
 

--- a/src/xbm/xbm.c
+++ b/src/xbm/xbm.c
@@ -6,6 +6,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <drm_fourcc.h>
 
 #include "common/dir.h"
 #include "common/grab-file.h"
@@ -26,7 +27,7 @@ texture_from_pixmap(struct wlr_renderer *renderer, struct pixmap *pixmap)
 	if (!pixmap) {
 		return NULL;
 	}
-	return wlr_texture_from_pixels(renderer, WL_SHM_FORMAT_ARGB8888,
+	return wlr_texture_from_pixels(renderer, DRM_FORMAT_ARGB8888,
 				       pixmap->width * 4, pixmap->width,
 				       pixmap->height, pixmap->data);
 }

--- a/src/xdg.c
+++ b/src/xdg.c
@@ -174,10 +174,10 @@ xdg_toplevel_view_close(struct view *view)
 }
 
 static void
-xdg_toplevel_view_for_each_popup(struct view *view,
+xdg_toplevel_view_for_each_popup_surface(struct view *view,
 		wlr_surface_iterator_func_t iterator, void *data)
 {
-	wlr_xdg_surface_for_each_popup(view->xdg_surface, iterator, data);
+	wlr_xdg_surface_for_each_popup_surface(view->xdg_surface, iterator, data);
 }
 
 static void
@@ -260,7 +260,7 @@ xdg_toplevel_view_unmap(struct view *view)
 static const struct view_impl xdg_toplevel_view_impl = {
 	.configure = xdg_toplevel_view_configure,
 	.close = xdg_toplevel_view_close,
-	.for_each_popup = xdg_toplevel_view_for_each_popup,
+	.for_each_popup_surface = xdg_toplevel_view_for_each_popup_surface,
 	.for_each_surface = xdg_toplevel_view_for_each_surface,
 	.map = xdg_toplevel_view_map,
 	.move = xdg_toplevel_view_move,

